Raylib-cs/types/Window.cs

```diff
@@ -0,0 +1,167 @@
+    {
+        return _instance ?? new Window(flags, width, height, title);
+    }
+    
+    /// <summary> See <see cref="Raylib.WindowShouldClose"/> </summary>
+    public bool ShouldClose() => Raylib.WindowShouldClose();
+    
+    /// <summary> See <see cref="Raylib.CloseWindow"/> </summary>
+    public void Close() => Raylib.CloseWindow();
+    
+    /// <summary> See <see cref="Raylib.TakeScreenshot(string)"/> </summary>
+    public void TakeScreenshot(string path) => Raylib.TakeScreenshot(path);
+    
+    
+    /// <summary> See <see cref="Raylib.IsWindowReady"/> </summary>
+    public bool IsReady() => Raylib.IsWindowReady();
+    
+    /// <summary> See <see cref="Raylib.IsWindowFullscreen"/> </summary>
+    public bool IsFullscreen() => Raylib.IsWindowFullscreen();
+    
+    /// <summary> See <see cref="Raylib.IsWindowHidden"/> </summary>
+    public bool IsHidden() => Raylib.IsWindowHidden();
+    
+    /// <summary> See <see cref="Raylib.IsWindowMinimized"/> </summary>
+    public bool IsMinimized() => Raylib.IsWindowMinimized();
+    
+    /// <summary> See <see cref="Raylib.IsWindowMaximized"/> </summary>
+    public bool IsMaximized() => Raylib.IsWindowMaximized();
+    
+    /// <summary> See <see cref="Raylib.IsWindowFocused"/> </summary>
+    public bool IsFocused() => Raylib.IsWindowFocused();
+    
+    /// <summary> See <see cref="Raylib.IsWindowResized"/> </summary>
+    public bool IsResized() => Raylib.IsWindowResized();
+    
+    /// <summary> See <see cref="Raylib.IsWindowState"/> </summary>
+    public bool IsState(ConfigFlags state) => Raylib.IsWindowState(state);
+    
+    
+    /// <summary> See <see cref="Raylib.SetWindowState"/> </summary>
+    public bool SetState(ConfigFlags state) => Raylib.SetWindowState(state);
+    
+    /// <summary> See <see cref="Raylib.ClearWindowState"/> </summary>
+    public void ClearState(ConfigFlags state) => Raylib.ClearWindowState(state);
+    
+    /// <summary> See <see cref="Raylib.ToggleFullscreen"/> </summary>
+    public void ToggleFullscreen() => Raylib.ToggleFullscreen();
+    
+    /// <summary> See <see cref="Raylib.MaximizeWindow"/> </summary>
+    public void Maximize() => Raylib.MaximizeWindow();
+    
+    /// <summary> See <see cref="Raylib.MinimizeWindow"/> </summary>
+    public void Minimize() => Raylib.MinimizeWindow();
+    
+    /// <summary> See <see cref="Raylib.RestoreWindow"/> </summary>
+    public void Restore() => Raylib.RestoreWindow();
+    
+    /// <summary> See <see cref="Raylib.SetWindowIcon"/> </summary>
+    public void SetIcon(Image image) => Raylib.SetWindowIcon(image);
+    
+    /// <summary> See <see cref="Raylib.SetWindowIcons"/> </summary>
+    public unsafe void SetIcons(Image* images, int count) => Raylib.SetWindowIcons(images, count);
+    
+    /// <summary> See <see cref="Raylib.SetWindowTitle(string)"/> </summary>
+    public void SetTitle(string title) => Raylib.SetWindowTitle(title);
```

Comment:
Could we change the setters and getters to properties. for example:
```c#
public bool IsReady => Raylib.IsWindowReady();

public string Title
{
    set => Raylib.SetWindowTitle(value);
}

public int Monitor
{
    get => Raylib.GetCurrentMonitor();
    set => Raylib.SetWindowMonitor(value);
}

Reply:
I could change to it.